ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+        estat = MM_ISExtendedNameBD_XP(data_base_XP->pField[i].FieldName);
+        if (estat == NM_CLASSICAL_DBF_AND_VALID_NAME ||
+            estat == MM_DBF_NAME_LOWERCASE_AND_VALID)
+        {
+            j = (short)strlen(data_base_XP->pField[i].FieldName);
+
+            retorn_fwrite = fwrite_function(&data_base_XP->pField[i].FieldName,
+                                            1, j, data_base_XP->pfDataBase);
+            if (retorn_fwrite != (size_t)j)
+            {
+                return FALSE;
+            }
+            MM_InitializeOffsetExtendedFieldNameFields(data_base_XP, i);
+            MM_InitializeBytesExtendedFieldNameFields(data_base_XP, i);
+        }
+        else if (estat == MM_VALID_EXTENDED_DBF_NAME)
+        {
+            if (*(data_base_XP->pField[i].ClassicalDBFFieldName) == '\0')
+            {
+                char nom_temp[MM_MAX_LON_FIELD_NAME_DBF];
+
+                MM_strnzcpy(nom_temp, data_base_XP->pField[i].FieldName,
+                            MM_MAX_LON_FIELD_NAME_DBF);
+                MM_ReturnValidClassicDBFFieldName(nom_temp);
+                nom_temp[MM_MAX_LON_CLASSICAL_FIELD_NAME_DBF - 1] = '\0';
+                if ((MM_CheckClassicFieldNameEqual(data_base_XP, nom_temp)) ==
+                    TRUE)
+                {
+                    char *c;
+
+                    c = MM_SetSubIndexFieldNam(
+                        nom_temp, i, MM_MAX_LON_CLASSICAL_FIELD_NAME_DBF);
+
+                    j = 0;
+                    while (MM_CheckClassicFieldNameEqual(data_base_XP, c) ==
+                               TRUE &&
+                           j < data_base_XP->nFields)
+                    {
+                        free_function(c);
+                        c = MM_SetSubIndexFieldNam(
+                            nom_temp, ++j, MM_MAX_LON_CLASSICAL_FIELD_NAME_DBF);
+                    }
+
+                    strcpy(data_base_XP->pField[i].ClassicalDBFFieldName, c);
+                    free_function(c);
+                }
+                else
+                    strcpy(data_base_XP->pField[i].ClassicalDBFFieldName,
+                           nom_temp);
+            }
+            j = (short)strlen(data_base_XP->pField[i].ClassicalDBFFieldName);
+
+            retorn_fwrite =
+                fwrite_function(&data_base_XP->pField[i].ClassicalDBFFieldName,
+                                1, j, data_base_XP->pfDataBase);
+            if (retorn_fwrite != (size_t)j)
+            {
+                free_function(zero);
+                return FALSE;
+            }
+
+            name_size =
+                MM_CalculateBytesExtendedFieldName(data_base_XP->pField + i);
+            MM_EscriuOffsetNomEstesBD_XP(data_base_XP, i, bytes_acumulats);
+            bytes_acumulats += name_size;
+        }
+        else
+        {
+            free_function(zero);
+            return FALSE;
+        }
+
+        if (fwrite_function(zero, 1, 11 - j, data_base_XP->pfDataBase) !=
```

Comment:
oh I now get it. Here's the padding. (It would have probably be clearer to copy the field name in a fixed-size null-initialized buffer and write it at once, instead to do it in 2 writes)

Reply:
There are some codes that I got from the MiraMon code assuming they are working fine.
Let's leave like that, then?